client/starwhale/api/_impl/service/service.py

```diff
@@ -149,8 +150,21 @@ def spec() -> t.Union[ServiceSpec, None]:
                 methods=["POST"],
             )
 
-        def index(opt: t.Any) -> FileResponse:
-            return FileResponse(os.path.join(STATIC_DIR_DEV, "client/index.html"))
+        def index(_: t.Any) -> HTMLResponse:
+            with open(os.path.join(STATIC_DIR_DEV, "client/index.html"), "r") as file:
+                content = file.read()
+            # https://github.com/star-whale/starwhale/pull/2996
+            root = os.getenv("SW_ONLINE_SERVING_ROOT_PATH", "")
+            opt = {
+                "root": root,
+            }
+            script = f"""<script type="text/javascript">
+                window.starwhaleConfig = {json.dumps(opt)}
+                </script>"""
+            content = content.replace("<!-- starwhale-client-placeholder -->", script)
+            # replace assets path
+            content = content.replace("/assets/", f"{root}/assets/")
```

Comment:
If user access to the service by ip directlly this still makes resources points to proxy backed urls? Can we add a header flag to make the api server aware of wether it is behind a proxy or not.